WIP fleet management documentation (#705)
neo4j/docs-aura: modules/ROOT/pages/fleet-management/data.adoc

= Data transparency

> This document describes the data structures used in fleet management messages.

Reviewer: ```suggestion
This page describes the data structures used in fleet management messages.
```

> == ConnectMessage

Reviewer: ```suggestion
== `ConnectMessage`
```

Message sent from the server to the Fleet Management service to establish a connection

> [cols="1,1,1"]

Reviewer: ```suggestion
[options="header", cols="1,1,1"]
```

|===
|Field |Type |Description
|server_id
|String
|Unique identifier for the server
|name
|String
|Name of the server
|dbms_id
|String

> |Unique identifier for the DBMS instance

Author: we probably should avoid using the word `instance` if possible.

Reviewer 2: Resolved

> |server_id
> |String
> |Unique identifier for the server
> |name
> |String
> |Name of the server
> |dbms_id
> |String
> |Unique identifier for the DBMS instance
> |server_version
> |String
> |Version of the server
> |project_id
> |String
> |Identifier for the project
> |plugin_version
> |String
> |Version of the plugin
> |build_profile
> |String
> |Build profile of the plugin

Reviewer: ```suggestion
|`server_id`
| String
| Unique identifier for the server
| `name`
| String
| Name of the server
| `dbms_id`
| String
| Unique identifier for the DBMS instance
| `server_version`
| String
| Version of the server
| `project_id`
| String
|I dentifier for the project
| `plugin_version`
| String
| Version of the plugin
| `build_profile`
| String
| Build profile of the plugin
```

|===

> == MetricsMessage

Reviewer: ```suggestion
== `MetricsMessage`
```

> Message sent from the server to the Fleet Management service containing metrics data

Author: ```suggestion
Message sent from the fleet management plugin to the Neo4j Aura endpoint containing metrics data
```
From server to service is less clear IMO. (several ocurrencies below)

Reviewer 2: Yeah I saw some things in here that I wanted to change, but I didn't want to edit the generated data. I made an exception for the page title.

Reviewer 2: Resolved

> [cols="1,1,1"]

Reviewer: ```suggestion
[options="header", cols="1,1,1"]
```

|===
|Field |Type |Description

> |timestamp
> |long
> |Unix timestamp in milliseconds when the message was created
> |project_id
> |String
> |Identifier for the project
> |dbms_id
> |String
> |Unique identifier for the DBMS instance
> |server_id
> |String
> |Unique identifier for the server
> |metrics
> |Map<String, List<DataPoint>>
> |Map of metric names to lists of DataPoint objects. Values are dynamically populated from the Fleet Management server.
> |metrics<>[].tags
> |Map<String, String>
> |Map of string key-value pairs providing additional context for the metric
> |metrics<>[].value
> |Double
> |Double value representing the metric measurement

Reviewer: ```suggestion
| `timestamp`
| long
| Unix timestamp in milliseconds when the message was created
| `project_id`
| String
| Identifier for the project
| `dbms_id`
| String
| Unique identifier for the DBMS instance
| `server_id`
| String
| Unique identifier for the server
| `metrics`
| Map<String, List<DataPoint>>
| Map of metric names to lists of DataPoint objects. Values are dynamically populated from the Fleet Management server.
| `metrics<>[].tags`
| Map<String, String>
| Map of string key-value pairs providing additional context for the metric
| `metrics<>[].value`
| Double
|Double value representing the metric measurement
```

|===

> == Neo4jConfigMessage

Reviewer: ```suggestion
== `Neo4jConfigMessage`
```

> Message sent from the server to the Fleet Management service containing neo4j config data

Reviewer: ```suggestion
Message sent from the server to the Fleet Management service containing neo4j configuration data.
```

> [cols="1,1,1"]

Reviewer: ```suggestion
[options="header", cols="1,1,1"]
```

|===
|Field |Type |Description

> |timestamp
> |long
> |Unix timestamp in milliseconds when the message was created
> |project_id
> |String
> |Identifier for the project
> |dbms_id
> |String
> |Unique identifier for the DBMS instance
> |server_id
> |String
> |Unique identifier for the server
> |neo4j_config
> |Map<String, Object>
> |Map of neo4j config names to config values. Values are dynamically populated from the Fleet Management server.

Reviewer: ```suggestion
| `timestamp`
| long
| Unix timestamp in milliseconds when the message was created
| `project_id`
| String
| Identifier for the project
| `dbms_id`
| String
| Unique identifier for the DBMS instance
| `server_id`
| String
| Unique identifier for the server
| `neo4j_config`
| Map<String, Object>
| Map of neo4j config names to config values. Values are dynamically populated from the Fleet Management server.
```

|===

> == PingMessage

Reviewer: ```suggestion
== `PingMessage`
```

> Message sent from the server to the Fleet Management service to retrieve configuration updates

Reviewer: ```suggestion
Message sent from the server to the Fleet Management service to retrieve configuration updates.
```

> [cols="1,1,1"]

Reviewer: ```suggestion
[options="header", cols="1,1,1"]
```

|===
|Field |Type |Description

> |server_id
> |String
> |Unique identifier for the server
> |server_version
> |String
> |Version of the server
> |project_id
> |String
> |Identifier for the project

Reviewer: ```suggestion
| `server_id`
| String
| Unique identifier for the server
| `server_version`
| String
| Version of the server
| `project_id`
| String
| Identifier for the project
```

|===

> == ReportingMessage

Reviewer: ```suggestion
== `ReportingMessage`
```

> Message sent from the server to the Fleet Management service containing reporting data

Reviewer: ```suggestion
Message sent from the server to the Fleet Management service containing reporting data.
```

> [cols="1,1,1"]

Reviewer: ```suggestion
[options="header", cols="1,1,1"]
```

|===
|Field |Type |Description
|timestamp
|long
|Unix timestamp in milliseconds when the message was created
|projectId
|String
|Identifier for the project
|dbms
|Dbms
|DBMS information object
|dbms.dbmsId
|String
|Unique identifier for the DBMS instance
|dbms.serverId
|String
|Unique identifier for the server
|dbms.databases
|List<String>
|List of database names
|dbms.edition
|String
|Edition of the DBMS (e.g., Enterprise, Community)
|dbms.packaging
|String
|Packaging of DBMS installation
|dbms.servers

> |List<Server>
> |List of servers in the DBMS cluster
> |dbms.servers[].serverId
> |String
> |Unique identifier for the server
> |dbms.servers[].name
> |String
> |Name of the server
> |dbms.servers[].address
> |String
> |Network address of the server
> |dbms.servers[].health
> |String
> |Current health status of the server
> |dbms.servers[].state
> |String
> |Current state of the server
> |dbms.servers[].modeConstraint
> |String
> |Mode constraint of the server
> |dbms.servers[].version
> |String
> |Neo4j version of the server
> |dbms.servers[].databases
> |List<Database>
> |List of databases hosted on this server
> |dbms.servers[].databases[].name
> |String
> |Name of the database
> |dbms.servers[].databases[].role
> |String
> |Role of the database in the cluster
> |dbms.servers[].databases[].type
> |String
> |Type of the database in the cluster
> |dbms.servers[].databases[].aliases
> |List<String>
> |List of database aliases
> |dbms.servers[].databases[].access
> |String
> |Access level of the database
> |dbms.servers[].databases[].databaseId
> |String
> |Unique identifier for the database
> |dbms.servers[].databases[].requestedStatus
> |String
> |Requested status of the database
> |dbms.servers[].databases[].currentStatus
> |String
> |Current status of the database
> |dbms.servers[].databases[].statusMessage
> |String
> |Status message describing the current state
> |dbms.servers[].databases[].default
> |boolean
> |Whether this is the default database
> |dbms.servers[].databases[].home
> |boolean
> |Whether this is the home database
> |dbms.servers[].databases[].currentPrimariesCount

Reviewer: ```suggestion
| `timestamp`
| long
| Unix timestamp in milliseconds when the message was created
| `projectId`
| String
| Identifier for the project
| `dbms`
| Dbms
| DBMS information object
| `dbms.dbmsId`
| String
| Unique identifier for the DBMS instance
| `dbms.serverId`
| String
| Unique identifier for the server
| `dbms.databases`
| List<String>
| List of database names
| `dbms.edition`
| String
| Edition of the DBMS (e.g., Enterprise, Community)
| `dbms.packaging`
| String
| Packaging of DBMS installation
| `dbms.servers`
| List<Server>
| List of servers in the DBMS cluster
| `dbms.servers[].serverId`
| String
| Unique identifier for the server
| `dbms.servers[].name`
| String
| Name of the server
| `dbms.servers[].address`
| String
| Network address of the server
| `dbms.servers[].health`
| String
| Current health status of the server
| `dbms.servers[].state`
| String
| Current state of the server
| `dbms.servers[].modeConstraint`
| String
| Mode constraint of the server
| `dbms.servers[].version`
| String
| Neo4j version of the server
| `dbms.servers[].databases`
| List<Database>
| List of databases hosted on this server
| `dbms.servers[].databases[].name`
| String
| Name of the database
| `dbms.servers[].databases[].role`
| String
| Role of the database in the cluster
| `dbms.servers[].databases[].type`
| String
| Type of the database in the cluster
| `dbms.servers[].databases[].aliases`
| List<String>
| List of database aliases
| `dbms.servers[].databases[].access`
| String
| Access level of the database
| `dbms.servers[].databases[].databaseId`
| String
| Unique identifier for the database
| `dbms.servers[].databases[].requestedStatus`
| String
| Requested status of the database
| `dbms.servers[].databases[].currentStatus`
| String
| Current status of the database
| `dbms.servers[].databases[].statusMessage`
| String
| Status message describing the current state
| `dbms.servers[].databases[].default`
| boolean
| Whether this is the default database
| `dbms.servers[].databases[].home`
| boolean
| Whether this is the home database
| `dbms.servers[].databases[].currentPrimariesCount`
```

|Integer
|Current number of primaries for this database
|dbms.servers[].databases[].currentSecondariesCount
|Integer
|Current number of secondaries for this database
|dbms.servers[].databases[].requestedPrimariesCount
|Integer
|Requested number of primaries for this database
|dbms.servers[].databases[].requestedSecondariesCount
|Integer
|Requested number of secondaries for this database
|dbms.servers[].databases[].creationTime
|long
|Timestamp when the database was created
|dbms.servers[].databases[].lastStartTime
|long
|Timestamp when the database was last started
|dbms.servers[].databases[].store
|String
|Store format of the database
|dbms.servers[].databases[].writer
|boolean
|Whether this instance is a writer for this database
|dbms.servers[].databases[].lastCommittedTxn
|Integer
|Last committed transaction ID

> |dbms.servers[].databases[].replicationLag
> |Integer
> |Current replication lag
> |dbms.servers[].databases[].graphCount
> |GraphCount
> |Current graph counts for the database
> |dbms.servers[].databases[].graphCount.node_count
> |Long
> |Number of nodes in the graph
> |dbms.servers[].databases[].graphCount.relationship_count
> |Long
> |Number of relationships in the graph
> |dbms.servers[].databases[].graphShards
> |List<String>
> |List of graph shards of this database
> |dbms.servers[].databases[].propertyShards
> |List<String>
> |List of property shards of this database
> |dbms.servers[].license
> |License
> |Main license information
> |dbms.servers[].license.type
> |String
> |Type of the license
> |dbms.servers[].license.state
> |String
> |Current state of the license
> |dbms.servers[].license.daysLeftOnTrial
> |Integer
> |Number of days remaining in trial period
> |dbms.servers[].license.totalTrialDays
> |Integer
> |Total number of days in trial period
> |dbms.servers[].bloomLicense
> |License
> |Bloom license information
> |dbms.servers[].bloomLicense.type
> |String
> |Type of the license
> |dbms.servers[].bloomLicense.state
> |String
> |Current state of the license
> |dbms.servers[].bloomLicense.daysLeftOnTrial
> |Integer
> |Number of days remaining in trial period
> |dbms.servers[].bloomLicense.totalTrialDays
> |Integer
> |Total number of days in trial period
> |dbms.servers[].gdsLicense
> |License
> |GDS license information
> |dbms.servers[].gdsLicense.type
> |String
> |Type of the license
> |dbms.servers[].gdsLicense.state
> |String
> |Current state of the license
> |dbms.servers[].gdsLicense.daysLeftOnTrial
> |Integer
> |Number of days remaining in trial period

Reviewer: ```suggestion
| Integer
| Current number of primaries for this database
| `dbms.servers[].databases[].currentSecondariesCount`
| Integer
| Current number of secondaries for this database
| `dbms.servers[].databases[].requestedPrimariesCount`
| Integer
| Requested number of primaries for this database
| `dbms.servers[].databases[].requestedSecondariesCount`
| Integer
| Requested number of secondaries for this database
| `dbms.servers[].databases[].creationTime`
| long
| Timestamp when the database was created
| `dbms.servers[].databases[].lastStartTime`
| long
| Timestamp when the database was last started
| `dbms.servers[].databases[].store`
| `String`
| Store format of the database
| `dbms.servers[].databases[].writer`
| boolean
| Whether this instance is a writer for this database
| `dbms.servers[].databases[].lastCommittedTxn`
| Integer
| Last committed transaction ID
| `dbms.servers[].databases[].replicationLag`
| Integer
| Current replication lag
| `dbms.servers[].databases[].graphCount`
| GraphCount
| Current graph counts for the database
| `dbms.servers[].databases[].graphCount.node_count`
| Long
| Number of nodes in the graph
| `dbms.servers[].databases[].graphCount.relationship_count`
| Long
| Number of relationships in the graph
| `dbms.servers[].databases[].graphShards`
| List<String>
| List of graph shards of this database
| `dbms.servers[].databases[].propertyShards`
| List<String>
| List of property shards of this database
| `dbms.servers[].license`
| License
| Main license information
| `dbms.servers[].license.type`
| String
| Type of the license
| `dbms.servers[].license.state`
| String
| Current state of the license
| `dbms.servers[].license.daysLeftOnTrial`
| Integer
| Number of days remaining in trial period
| `dbms.servers[].license.totalTrialDays`
| Integer
| Total number of days in trial period
| `dbms.servers[].bloomLicense`
| License
| Bloom license information
| `dbms.servers[].bloomLicense.type`
| String
| Type of the license
| `dbms.servers[].bloomLicense.state`
| String
| Current state of the license
| `dbms.servers[].bloomLicense.daysLeftOnTrial`
| Integer
| Number of days remaining in trial period
| `dbms.servers[].bloomLicense.totalTrialDays`
| Integer
| Total number of days in trial period
| `dbms.servers[].gdsLicense`
| License
| GDS license information
| `dbms.servers[].gdsLicense.type`
| String
| Type of the license
| `dbms.servers[].gdsLicense.state`
| String
| Current state of the license
| `dbms.servers[].gdsLicense.daysLeftOnTrial`
| Integer
| Number of days remaining in trial period
```

|dbms.servers[].gdsLicense.totalTrialDays
|Integer
|Total number of days in trial period
|dbms.servers[].plugins
|List<Plugin>
|List of plugins installed on the server
|dbms.servers[].plugins[].filename
|String
|Filename of the plugin
|dbms.servers[].plugins[].name
|String
|Manifest name of the plugin, if available
|dbms.servers[].plugins[].version
|String
|Manifest version of the plugin, if available
|dbms.servers[].plugins[].vendor
|String
|Manifest vendor of the plugin, if available
|pluginVersion
|String
|Version of the fleet management plugin
|osName
|String
|Operating system name
|osVersion
|String
|Operating system version
|osArch
|String
|Operating system architecture
|jvmVersion
|String
|Java Virtual Machine version
|jvmVendor
|String

> |Java Virtual Machine vendor

Reviewer: ```suggestion
| `dbms.servers[].gdsLicense.totalTrialDays`
| Integer
| Total number of days in trial period
| `dbms.servers[].plugins`
| List<Plugin>
| List of plugins installed on the server
| `dbms.servers[].plugins[].filename`
| String
| Filename of the plugin
| `dbms.servers[].plugins[].name`
| String
| Manifest name of the plugin, if available
| `dbms.servers[].plugins[].version`
| String
| Manifest version of the plugin, if available
| `dbms.servers[].plugins[].vendor`
| String
| Manifest vendor of the plugin, if available
| `pluginVersion`
| String
| Version of the fleet management plugin
| `osName`
| String
| Operating system name
| `osVersion`
| String
| Operating system version
| `osArch`
| String
| Operating system architecture
| `jvmVersion`
| String
| Java Virtual Machine version
| `jvmVendor`
| String
| Java Virtual Machine vendor
```
